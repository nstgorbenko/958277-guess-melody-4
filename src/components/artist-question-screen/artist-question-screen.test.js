import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from './artist-question-screen.jsx';

const testQuestion = {
  type: `artist`,
  song: {
    artist: `Jim Beam`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [{
    picture: `https://api.adorable.io/avatars/128/0`,
    artist: `John Snow`,
  }, {
    picture: `https://api.adorable.io/avatars/128/1`,
    artist: `Jack Daniels`,
  }, {
    picture: `https://api.adorable.io/avatars/128/2`,
    artist: `Jim Beam`,
  }],
};

describe(`ArtistQuestionScreen Component rendering`, () => {
  it(`ArtistQuestionScreen is rendered correctly`, () => {
    const tree = renderer
      .create(
          <ArtistQuestionScreen
            question={testQuestion}
            onAnswer={() => {}}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
