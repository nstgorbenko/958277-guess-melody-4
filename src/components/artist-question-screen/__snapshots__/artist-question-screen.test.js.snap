// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen Component rendering ArtistQuestionScreen is rendered correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
        />
      </div>
    </div>
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="artist-0"
      >
        <img
          alt="John Snow"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/0"
        />
        John Snow
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="artist-1"
      >
        <img
          alt="Jack Daniels"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/1"
        />
        Jack Daniels
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="artist-2"
      >
        <img
          alt="Jim Beam"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/2"
        />
        Jim Beam
      </label>
    </div>
  </form>
</section>
`;
